File: CPP/101.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        // Scan the nodes in opposite ways
        if (!root) return true;
        // Tree is not empty
        else return isSame(root->left, root->right);
    }
    bool isSame(TreeNode *left, TreeNode *right){
        // In case both left and right are empty which is true
        if (!(left && right)) return left == right;
        // Left and Right are both not empty
        return ((left->val == right->val) && (isSame(left->right, right->left)) && (isSame(left->left, right->right)));
    }
};

File: Python/101.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
def isSame(left: TreeNode, right: TreeNode):
    if not (left and right): # When either one is empty
        return left == right # In case both are empty
    else:
        return ((left.val == right.val) and (isSame(left.right, right.left)) and (isSame(left.left, right.right)))

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        if not root: return True
        else: return isSame(root.left, root.right)
